File: src/main/java/de/dp_coding/zammadplugin/ZammadSettingsAction.java
package de.dp_coding.zammadplugin;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.Project;
import de.dp_coding.zammadplugin.ui.ZammadSettingsDialog;
import org.jetbrains.annotations.NotNull;

/**
 * Action for configuring Zammad API settings.
 */
public class ZammadSettingsAction extends AnAction {
    
    @Override
    public void actionPerformed(@NotNull AnActionEvent e) {
        Project project = e.getProject();
        if (project == null) {
            return;
        }
        
        ZammadSettingsDialog dialog = new ZammadSettingsDialog(project);
        dialog.show();
    }
}

File: src/main/kotlin/de/dp_coding/zammadplugin/ZammadSettingsAction.kt
package de.dp_coding.zammadplugin

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import de.dp_coding.zammadplugin.ui.ZammadSettingsDialog

/**
 * Action for configuring Zammad API settings.
 */
class ZammadSettingsAction : AnAction() {
    
    override fun actionPerformed(e: AnActionEvent) {
        val project = e.project ?: return
        
        val dialog = ZammadSettingsDialog(project)
        dialog.show()
    }
}